### PR TITLE
Address some lint errors

### PR DESCRIPTION
**Pull request by trshafer**
Addresses some lint errors in the go report card.

### CODE REVIEW

File: apis/duck/v1alpha1/conditions_types.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func (t *KResource) Populate() {
 }
 
 // GetListType implements apis.Listable
-func (r *KResource) GetListType() runtime.Object {
+func (t *KResource) GetListType() runtime.Object {
```
Comment (evankanderson) [inline apis/duck/v1alpha1/conditions_types.go]: Why does this need `t`?

Comment (trshafer) [reply apis/duck/v1alpha1/conditions_types.go]: There is a lint test that errors when a receiver has a different name within the same file. The receiver is named `t` on line 177: https://github.com/knative/pkg/pull/351/files/f35a510212e1636b012c3bee893f6043c18bde95#diff-04abbb75ad1eb1e871f98943273acd40L177

Comment (mattmoor) [reply apis/duck/v1alpha1/conditions_types.go]: I think the question is why we need it at all?

File: apis/duck/v1alpha1/legacy_targetable_types.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func (t *LegacyTarget) Populate() {
 }
 
 // GetListType implements apis.Listable
-func (r *LegacyTarget) GetListType() runtime.Object {
+func (t *LegacyTarget) GetListType() runtime.Object {
```
Comment (mattmoor) [inline apis/duck/v1alpha1/legacy_targetable_types.go]: delete?

File: apis/duck/v1alpha1/retired_targetable_types.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ func (t *Target) Populate() {
 }
 
 // GetListType implements apis.Listable
-func (r *Target) GetListType() runtime.Object {
+func (t *Target) GetListType() runtime.Object {
```
Comment (mattmoor) [inline apis/duck/v1alpha1/retired_targetable_types.go]: delete?

File: apis/duck/v1alpha1/conditions_types.go (hunk 1 above)
Comment (trshafer) [reply apis/duck/v1alpha1/conditions_types.go]: Ah thanks for the clarification. My understanding is that this implements the Listable interface: https://github.com/trshafer/pkg/blob/f35a510212e1636b012c3bee893f6043c18bde95/apis/interfaces.go#L48. So the objects can be used here: https://github.com/trshafer/pkg/blob/f35a510212e1636b012c3bee893f6043c18bde95/apis/duck/typed.go#L49

Comment (mattmoor) [reply apis/duck/v1alpha1/conditions_types.go]: To further clarify, why not this?
```suggestion
func (*KResource) GetListType() runtime.Object {
```

File: apis/duck/v1alpha1/legacy_targetable_types.go (hunk 2 above)
Comment (mattmoor) [reply apis/duck/v1alpha1/legacy_targetable_types.go]:
```suggestion
func (*LegacyTarget) GetListType() runtime.Object {
```

File: apis/duck/v1alpha1/retired_targetable_types.go (hunk 3 above)
Comment (mattmoor) [reply apis/duck/v1alpha1/retired_targetable_types.go]:
```suggestion
func (*Target) GetListType() runtime.Object {
```

File: apis/duck/v1alpha1/conditions_types.go (hunk 1 above)
Comment (trshafer) [reply apis/duck/v1alpha1/conditions_types.go]: Great suggestion. Thanks.

**Pull request merged**